src/parsemultipart.jl

```diff
@@ -0,0 +1,52 @@
+function parse_multipart_chunk!(chunk, d)
+    re = r"(\n){2,}|(\r){2,}|[\r\n]{4,}"
+    Parsers.exec(re, chunk) || return nothing
+    i, j = re.ovec
+    description = String(chunk[1:i])
+    content = Parsers.nextbytes(re, chunk)
+
+    v = match(r"Content-Disposition: form-data; name=\"(.*)\"; filename=\"(.*)\"[\r\n]+Content-Type: (\S*)", description)
+    if v !== nothing
+        push!(d["name"], v[1])
+        push!(d["filename"], v[2])
+        push!(d["contenttype"], v[3])
+    else
+        v = match(r"Content-Disposition: form-data; name=\"(.*)\"", description)
+        v == nothing && return
```

Comment:
This `v == nothing && return` makes me uncomfortable because we've basically gotten into a bad state and we're just returning. I think we probably need to read up on the spec to see exactly what is and isn't required in terms of the `Content-Disposition: form-data;` tags and make sure we're following that. Is `name` _always required? What if just `name` and `Content-Type` are present? Seems like we want to make sure we're accounting for what the spec allows here.